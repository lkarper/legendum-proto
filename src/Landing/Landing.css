.Landing__image-container {
    margin: 15px;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.Landing__arrow-container {
    height: 20px;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

.Landing__arrow {
    width: 0;
    height: 0;
    border-width: 20px 20px 0px 20px;
    border-style: solid;
    border-color: black transparent transparent transparent;
    position: relative;
}

.Landing__arrow:after {
    border-color: white transparent transparent transparent;
    border-style: solid;
    border-width: 20px 20px 0px 20px;
    bottom: 1px;
    content: "";
    position: absolute;
    top: -23px;
    left: -20px;
}

.Landing__greeting-image {
    max-width: 75%;
}

.Landing__speech-bubble-container {
    width: 75%;
}

.Landing__speech-bubble {
    border: 2px solid black;
    background-color: white;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 10px;
}

.Landing__speech-text {
    font-size: 32px;
    margin: 20px 5px; 
    font-family: 'EB Garamond', serif;
}

.Landing__section {
    font-family: 'EB Garamond', serif;
}

.Landing__section .Landing__section {
    width: 100%;
}

.Landing__section p {
    font-size: 18px;
    line-height: 28px;
}

.Landing__link-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    border: 2px solid #FFC71F;
    background-color: #721D2B;
    padding: 5px;
    width: 90%;
    margin: 5px auto;
}

.Landing__link {
    color: #FFC71F;
    text-decoration: none;
    font-family: 'EB Garamond', serif;
    margin: 5px 10px;
    font-size: 20px;
    text-align: center;
}

.Landing__link:active {
    text-decoration: underline;
}

.Landing__link:hover {
    color: goldenrod;
}

.Landing__demo {
    display: block;
    margin: 10px auto;
    padding: 10px;
    font-size: 20px;
}

.Landing__p.error {
    color: darkred;
}

@media screen and (min-width: 1000px) {
    .Landing__greeting-image {
        max-width: 50%;
    }

    .Landing__link {
        font-size: 24px;
    }

    .Landing__arrow:after {
        top: -22px;
    }

    .Landing__link-container {
        width: 50%;
        margin-top: 15px;
    }

    .Landing__link {
        font-size: 20px;
    }
}
