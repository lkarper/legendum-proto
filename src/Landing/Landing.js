import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import TokenService from '../services/token-service';
import NextStory from '../NextStory/NextStory';
import './Landing.css';

const Landing = (props) => {
    return (
        <div className="Landing__container">
            <div className='Landing__image-container'>
                <div className="Landing__speech-bubble-container">
                    <div className="Landing__speech-bubble">
                        <p className='Landing__speech-text'>Welcome!</p>
                    </div>
                    <div aria-hidden='true' className='Landing__arrow-container'>
                        <div className='Landing__arrow left'></div>
                        <div className='Landing__arrow center'></div>
                        <div className='Landing__arrow right'></div>
                    </div>
                </div>
                <img
                    className='Landing__greeting-image' 
                    src='/images/enni.png'
                    alt='A friendly-looking, three-headed dog.'
                />
            </div>
            {!TokenService.hasAuthToken() ? 
                <div className='Landing__link-container'>
                    <Link
                        className='Landing__link'
                        to="/register"
                    >Register</Link>
                    <FontAwesomeIcon 
                        className='Landing__leaf' 
                        icon={['fab', 'pagelines']} 
                    />
                    <Link
                        className='Landing__link'
                        to="/game/story/1"
                    >Play without Registering</Link>
                    <FontAwesomeIcon 
                        className='Landing__leaf' 
                        icon={['fab', 'pagelines']} 
                    />
                    <Link
                        className='Landing__link'
                        to="/login"
                    >Login</Link>
                </div>
                : 
                <div className='Landing__link-container'>
                    <Link
                        className='Landing__link'
                        to='/dashboard'
                    >Go to Dashboard</Link>
                    <FontAwesomeIcon 
                        className='Landing__leaf' 
                        icon={['fab', 'pagelines']} 
                    />
                    <NextStory />
                </div>    
            }
        </div>
    )
}

export default Landing;