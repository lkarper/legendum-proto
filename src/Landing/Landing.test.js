import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Landing from './Landing';

describe('Landing component', () => {
  it('renders the UI without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const wrapper = mount(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
