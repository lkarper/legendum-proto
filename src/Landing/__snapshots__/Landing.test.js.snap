// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Landing>
      <div
        className="Landing__image-container"
      >
        <div
          className="Landing__speech-bubble-container"
        >
          <div
            className="Landing__speech-bubble"
          >
            <p
              className="Landing__speech-text"
            >
              Welcome!
            </p>
          </div>
          <div
            aria-hidden="true"
            className="Landing__arrow-container"
          >
            <div
              className="Landing__arrow left"
            />
            <div
              className="Landing__arrow center"
            />
            <div
              className="Landing__arrow right"
            />
          </div>
        </div>
        <img
          alt="A friendly-looking, three-headed dog."
          className="Landing__greeting-image"
          src="/images/enni.png"
        />
      </div>
      <section
        className="Landing__section"
      >
        <header>
          <h2>
            What is Legendum?
          </h2>
        </header>
        <p>
          Legendum is a Latin learning app that uses stories, images, and short quizzes to teach Latin!
        </p>
        <p>
          This is a demo version of Legendum designed to get initial feedback from users. The site is now hooked up to a backend server, so you can now create an account and save your own notes and progress!
        </p>
        <p>
          <Link
            to="/register"
          >
            <LinkAnchor
              href="/register"
              navigate={[Function]}
            >
              <a
                href="/register"
                onClick={[Function]}
              >
                Click here
              </a>
            </LinkAnchor>
          </Link>
           to create an account and get started!
        </p>
        <p>
          (You don't need to register to learn, but you'll need an account to save your own notes and your progress.)
        </p>
        <p>
          This demo version of Legendum contains two chapters.  The goal is to release new chapters regularly in the future.
        </p>
        <p>
          After you test the app, please fill out this 
          <a
            className="Landing__feedback-link"
            href="https://forms.gle/VrAvhA4pQ4SxARf89"
            rel="noopener noreferrer"
            target="_blank"
          >
            form
          </a>
          !
        </p>
        <p>
          Thank you! Thank you! Thank you!
        </p>
      </section>
      <section
        className="Landing__section"
      >
        <header>
          <h2>
            How does Legendum Work?
          </h2>
        </header>
        <p>
          Legendum was designed with two goals in mind: keep the learner engaged and build the learner's confidence.
        </p>
        <section
          className="Landing__section"
        >
          <header>
            <h3>
              Engagement Through Stories and Images
            </h3>
          </header>
          <p>
            Legendum is designed like a game and users learn Latin by completing chapters in a story. The story is full of fun characters, charming images, and the occasional bit of witty banter.
          </p>
          <p>
            Each chapter begins with a story in English that sets the scene.
          </p>
          <p>
            <Link
              to="/game/story/1"
            >
              <LinkAnchor
                href="/game/story/1"
                navigate={[Function]}
              >
                <a
                  href="/game/story/1"
                  onClick={[Function]}
                >
                  Click here
                </a>
              </LinkAnchor>
            </Link>
             to get started with Chapter 1!
          </p>
        </section>
        <section
          className="Landing__section"
        >
          <header>
            <h3>
              Build your Confidence with Comprehensible Input
            </h3>
          </header>
          <p>
            After the user reads through the story, the chapter's Latin lesson begins. Lessons are designed on the principles of Comprehensible Input. In short, proponents of this theory of language learning argue that the best way to learn a language is to be exposed to it without the filter of another language.
          </p>
          <p>
            Legendum uses short sentences, repetition, and images to help students understand the Latin that they are reading without having to translate it in their heads!
          </p>
          <p>
            There are, however, hints, explanations, and tips throughout the lesson that a user can toggle on and off. This means that users can avoid the stress of "not knowing" what's going on. Stress is a major impediment to language learning, so Legendum aims to make learning Latin as stress-free as possible!
          </p>
          <p>
            Registered users can save the grammatical tidbits that are presented and can even add personal notes.
          </p>
          <p>
            Too exicted about Latin to start with an English story? Jump right in and 
            <Link
              to="/game/exercises/1/learn"
            >
              <LinkAnchor
                href="/game/exercises/1/learn"
                navigate={[Function]}
              >
                <a
                  href="/game/exercises/1/learn"
                  onClick={[Function]}
                >
                  starting learning with Exercise One
                </a>
              </LinkAnchor>
            </Link>
            !
          </p>
        </section>
        <section
          className="Landing__section"
        >
          <header>
            <h3>
              Test yourself with short quizzes
            </h3>
          </header>
          <p>
            Each lesson is followed by a short quiz that helps users gauge their comprehension and retention.
          </p>
          <p>
            These quizzes are low-stress undertakings. If an incorrect answer is given, the user is presented with a short hint or explanation on why the answer is incorrect. Users can simply try again if they get the question wrong and overall "grades" are not kept. Registered users have access to their notes throughout.
          </p>
          <p>
            Already know some Latin and want to test yourself out right away? 
            <Link
              to="/game/exercises/1/do"
            >
              <LinkAnchor
                href="/game/exercises/1/do"
                navigate={[Function]}
              >
                <a
                  href="/game/exercises/1/do"
                  onClick={[Function]}
                >
                  Try the first quiz!
                </a>
              </LinkAnchor>
            </Link>
          </p>
        </section>
        <section
          className="Landing__section"
        >
          <header>
            <h3>
              Keep Track of your Progress
            </h3>
          </header>
          <p>
            Registered users can track their progress through lessons and stories, see how many times they've completed each chapter, and access and edit their notes on the 
            <Link
              to="/dashboard"
            >
              <LinkAnchor
                href="/dashboard"
                navigate={[Function]}
              >
                <a
                  href="/dashboard"
                  onClick={[Function]}
                >
                  dashboard
                </a>
              </LinkAnchor>
            </Link>
            .
          </p>
        </section>
      </section>
    </Landing>
  </Router>
</BrowserRouter>
`;
